rasa/core/policies/ensemble.py

```diff
@@ -556,6 +556,20 @@ def _is_form_policy(policy_name: Text) -> bool:
 
         return policy_name.endswith("_" + FormPolicy.__name__)
 
+    @staticmethod
+    def _prediction_type(predictions: Dict[Text, PolicyPrediction]) -> Optional[bool]:
+        # No user input prediction overrule all other predictions.
+        if any(
+            prediction.is_end_to_end_prediction is None
```

Comment:
@joejuzl Here we branch on `None` as `None` means it's a `loop` prediction

Reply:
ok so `use_only_end_to_end` is:
- `None` if at least one `prediction.is_end_to_end_prediction` is `None`
- `False` if all are `False`
- `True` if at least one is `True` and none are `None`.

Then, yes I do find that hard to follow.